Simplify public URL configuration (#3837)
thunder-id/thunderid: docs/content/guides/getting-started/configuration.mdx

| `server.hostname` | `localhost` | The hostname or IP address the server binds to |
| `server.port` | `8090` | Port the server listens on |
| `server.http_only` | `false` | If `true`, disables HTTPS and uses HTTP only (not recommended for production) |

> | `server.public_url` | _(derived from server hostname, port and scheme)_ | The public URL clients use to reach the server, if it differs from the bind address. Derived when unset. |

Reviewer: protocol is appropriate here instead of "scheme"

Author: Resolved in [Simplify URL configuration](https://github.com/thunder-id/thunderid/pull/3837/commits/252c2af58341aa8066d54d33d0c652b028523275)

| `server.identifier` | `default-deployment` | Unique identifier for this deployment instance |

> When `public_url` is set, the `gate_client` settings below default to it, so most deployments only need to configure the server section above.

Reviewer: ```suggestion
When `server.public_url` is set, the `gate_client` settings below default to the corresponding hostname, port, and protocol from that URL. Most deployments only need to configure the server section.
```

## Gate Client Configuration

> Configures the connection to <ProductName /> Gate (the login UI). Every setting is optional. When unset, the `gate_client` settings default to the server's public URL, so this section is only needed when Gate runs separately from the server.

Reviewer: ```suggestion
Configures the connection to <ProductName /> Gate (the login UI). Every setting is optional. By default, the `gate_client` settings are derived from `server.public_url`, so you only need to configure this section when Gate is hosted separately from the server.
```
